system/jlib/jexcept.cpp

```diff
@@ -1737,5 +1737,60 @@ void IErrorReceiver::reportWarning(WarnErrorCategory category, int warnNo, const
     report(warn);
 }
 
+LogMsgCategory mapToLogMsgCategory(ErrorSeverity severity, MessageAudience aud)
```

Comment:
Not sure about this function for a couple of reasons.  It returns an object rather than a reference to an object, and it is arguably easier to create a local variable.

E.g.
LogMsgClass getLoggingClass(severity) ....
LogMsgCategory category(aud, getLoggingClass(severity));

Alternatively return a const LogMsgCategory & from the function.